gpii/node_modules/processReporter/README.md

```diff
@@ -0,0 +1,85 @@
+
+### Process Reporter API
+
+The app currently supports the following urls:
+
+    {urlToProcessReporterServer}/processes            // GET ONLY
+    {urlToProcessReporterServer}/processes/{solution} // GET ONLY
+
+The first form cycles through the solutions registry, and for each entry, determines if the associated processes are running.  The second form reports only the solution requested, for example:
+
+    {urlToProcessReporterServer}/processes/org.gnome.desktop.a11y.magnifier
+
+In both cases, the Process Reporter uses the information stored in the `isRunning` block of a solutions registry entry.  For example, the `isRunning` block for the Orca screen reader (the solution entry `org.gnome.orca`) is as
+follows:
+```
+"isRunning": [
+    {
+        "type": "gpii.processReporter.find",
+        "command": "orca"
+    }
+]
+```
+
+The payload returned is an array of solutions and their run state, followed by the platform:
+```
+{
+    "solutions": [
+        {
+            "id": "org.gnome.desktop.a11y.applications",
+            "running": true
+        },
+        {
+            "id": "org.gnome.orca",
+            "running": false
+        },
+        ...
+    ],
+    "OS":
+    {
+        "id": "linux",
+        "version": "4.5.7-300.fc24.x86-64"
+}}
+```
+
+If the status of only one solution is requested, then the payload is information about that solution:
+
+```
+{
+    "solutions":
+    {
+        "id": "org.gnome.orca",
+        "running":false
+    },
+    "OS":
+    {
+        "id": "linux",
+        "version": "4.5.7-300.fc24.x86-64"
+    }
+}
+```
+
+### Special cases
+
+If the solutions registry entry does not have an `isRunning` block, then the solution is skipped and nothing is reported about its run-state.  NOTE: This may be incorrect, and is currently under review.
+
+Most solutions run-state is based on a corresponding entry in the underlying OS's process or tasklist table.  For example, the Linux/GNOME solution `org.gnome.orca` is running if the `orca` screen reader process is running.  Similarly, on Windows, `org.nvda-project` is running if there is an `nvda` process in the tasklist.  Some solutions, however, are not strictly process-oriented.  An example is the GNOME Shell screen magnifier (solution `org.gnome.desktop.a11y.magnifier`).  The magnifier is not a separate process, but is built into GNOME Shell.  Thus, the magnifier is running if (1) the `gnome-shell` process is running and (2) the `org.gnome.desktop.a11y.applications screen-magnifier-enabled` GSetting is `true`.  The Linux/GNOME implementation of the process reporter optionally checks the state of the setting, and the `isRunning` block contains that information:
+
+```
+"isRunning": [
+    {
+        "type": "gpii.processReporter.find",
+        "command": "gnome-shell",
+        "schema": "org.gnome.desktop.a11y.applications",
+        "setting": "screen-magnifier-enabled"
+    }
```

Comment:
Do you know what the approach is in the DR if multiple entries are listed in "isInstalled"? If they're AND'ing the blocks (ie. meaning all of them have to be true), it might be worth considering doing here as well... It seems cleaner to me to have one block of type "gpii.processReporter.find" and then another with another type related to checking the gsettings